src/redux/Books/Books.js

```diff
@@ -0,0 +1,54 @@
+const ADD_BOOK = 'ADD_BOOK';
+const REMOVE_BOOK = 'REMOVE_BOOK';
+
+const initialState = [
+  {
+    title: 'The Hunger Games',
+    author: 'Author1',
+    id: 1,
+  },
+
+  {
+    title: 'Dune',
+    author: 'Author2',
+    id: 2,
+  },
+];
+
+export const booksReducer = (state = initialState, action) => {
+  switch (action.type) {
+    case ADD_BOOK:
+      return [
+        ...state,
+        {
+          id: 1,
+          title: action.title,
+          author: 'action.author',
+        },
+      ];
```

Comment:
- [ ] In this case, you are hardcoding the id of the book and author, as author is in brackets. I suggest to update state dynamically without hardcoded values. 
```
#REDUCER: 
case ADD_BOOK:
 { return [...state, action.payload]; }
```
```
#ACTION: 
export const addBook = (payload) => async (dispatch) => {
  await postBook(payload);
  dispatch({ type: ADD_BOOK, payload });
};
```
```
# COMPONENT:
  const submitBookToStore = (e) => {
    e.preventDefault();
    const newBook = {
      title,
      item_id: Math.round(Math.random() * 1000),
      category,
    };
    dispatch(addBook(newBook));
    setTitle('');
  };
```

As you can see in this example, the new book object is created in controller and passed to action as a payload. Try to do the same 👍🏻